src/test/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchProcessResponseFilesImplTest.java

```diff
@@ -0,0 +1,256 @@
+
+    @AfterAll
+    public static void close() {
+        EOD_HELPER_MOCKS.close();
+    }
+
+    @BeforeEach
+    void setUp() {
+        detsToArmBatchProcessResponseFilesImpl = new DetsToArmBatchProcessResponseFilesImpl(
+            externalObjectDirectoryRepository,
+            armDataManagementApi,
+            fileOperationService,
+            armDataManagementConfiguration,
+            objectMapper,
+            userIdentity,
+            timeHelper,
+            externalObjectDirectoryService,
+            logApi,
+            configuration,
+            osrRepository
+        );
+
+        String batchMetadataFilename = "dropzone/DARTS/response/DETS_a17b9015-e6ad-77c5-8d1e-13259aae1895_6a374f19a9ce7dc9cc480ea8d4eca0fb_1_iu.rsp";
+        batchInputUploadFileFilenameProcessor = new BatchInputUploadFileFilenameProcessor(batchMetadataFilename);
+        String createRecordFilename = "dropzone/DARTS/response/6a374f19a9ce7dc9cc480ea8d4eca0fb_12374f19a9ce7dc9cc480ea8d4eca0fb_1_cr.rsp";
+        createRecordFilenameProcessor = new CreateRecordFilenameProcessor(createRecordFilename);
+        String uploadFileFilename = "dropzone/DARTS/response/6a374f19a9ce7dc9cc480ea8d4eca0fb_45674f19a9ce7dc9cc480ea8d4eca0fb_1_iu.rsp";
+        uploadFileFilenameProcessor = new UploadFileFilenameProcessor(uploadFileFilename);
+        String invalidLineFileFilename = "dropzone/DARTS/response/6a374f19a9ce7dc9cc480ea8d4eca0fb_78974f19a9ce7dc9cc480ea8d4eca0fb_1_il.rsp";
+        invalidLineFileFilenameProcessor = new InvalidLineFileFilenameProcessor(invalidLineFileFilename);
+
+        externalObjectDirectoryEntity = new ExternalObjectDirectoryEntity();
+        externalObjectDirectoryEntity.setId(1);
+        externalObjectDirectoryEntity.setStatus(EodHelper.armProcessingResponseFilesStatus());
+        externalObjectDirectoryEntity.setTransferAttempts(1);
+        objectStateRecordEntity = new ObjectStateRecordEntity();
+        objectStateRecordEntity.setUuid(123L);
+        externalObjectDirectoryEntity.setOsrUuid(objectStateRecordEntity.getUuid());
+
+    }
+
+    @Test
+    void getManifestFilePrefix() {
+        // given
+        when(configuration.getDetsManifestFilePrefix()).thenReturn("DETS");
+
+        // when
+        String result = detsToArmBatchProcessResponseFilesImpl.getManifestFilePrefix();
+
+        // then
+        assertNotNull(result);
+        verify(configuration, times(1)).getDetsManifestFilePrefix();
+    }
+
+    @Test
+    void preProcessResponseFilesActions() {
+        // given
+        when(osrRepository.findByArmEodId(String.valueOf(externalObjectDirectoryEntity.getId()))).thenReturn(Optional.of(objectStateRecordEntity));
+
+        // when
+        detsToArmBatchProcessResponseFilesImpl.preProcessResponseFilesActions(externalObjectDirectoryEntity.getId());
+
+        // then
+        verify(osrRepository, times(1)).findByArmEodId(String.valueOf(externalObjectDirectoryEntity.getId()));
```

Comment:
Verifications are usually better when they use hard coded values instead of method calls.

I.e getId() could return null but we actually want it to be 123. So checking 123 gives us more assurance

Reply:
done